src/LuaObject.h

```diff
@@ -74,18 +74,29 @@ struct SerializerPair {
 	typedef std::string (*Serializer)(LuaWrappable *o);
 	typedef bool (*Deserializer)(const char *stream, const char **next);
 
+	typedef void (*ToJson)(Json::Value &out, LuaWrappable *o);
+	typedef bool (*FromJson)(const Json::Value &obj);
+
 	SerializerPair() :
 		serialize(nullptr),
-		deserialize(nullptr)
+		deserialize(nullptr),
```

Comment:
I'm confused. Why do we need to have 4 functions now?

Reply:
This PR does not remove the current serialisation (read/unserialize) codepath because it still supports reading files that are written with that format.

I haven't removed the write path for it because I wanted it to be easy to dual-write both old & new formats which would make new saves backwards compatible. However I haven't actually implemented dual-write so if we don't want that then the bytestream write path can be deleted.

Reply:
Also for more context, I do expect to break old saves "soonish" if I proceed with (a) [serializing to CBOR instead of JSON](https://pioneerspacesim.net/forum/viewtopic.php?f=7&p=5361#p5361), and (b) changing Lua side structures like Equipment to stop embedding so much "static" game data into saves (as you suggested in the [forum thread](https://pioneerspacesim.net/forum/viewtopic.php?f=3&t=531))

*But* I don't want to prematurely break saves before I'm confident of completing at least the JSON->CBOR change.